redis/src/main/scala/zio/redis/RespValue.scala

```diff
@@ -146,16 +150,24 @@ object RespValue {
               case state              => CollectingArray(rem, vals, state.feed)
             }
 
-          case ExpectingBulk => Done(bulkString(line))
-          case _             => Failed
+          case CollectingBulkString(rem, vals, index) =>
+            val allLineBytes = Chunk.fromArray(line.getBytes(StandardCharsets.UTF_8))
+            val lineBytes    = if (allLineBytes.length > rem) allLineBytes.slice(0, rem) else allLineBytes
+            val bytesBuilder = if (index > 0) vals ++= (Lf +: lineBytes) else vals ++= lineBytes
+
+            if (rem - lineBytes.length > 0)
+              CollectingBulkString(rem - lineBytes.length - 1, bytesBuilder, index + lineBytes.length)
```

Comment:
I'd factor out these length calls.

Reply:
I have updated it to simplify the CollectinngBulkString logic.
  As stated in redis RESP spec about arrays, I think we can replace Transduce.splitLines for Transduce.splitOn(CrLf),
 so unnecessary complications are avoided when it comes to recomposing the possible value of the bulk string.